ghostwriter/api/views.py

```diff
@@ -577,6 +581,89 @@ def post(self, request, *args, **kwargs):
         return JsonResponse(utils.generate_hasura_error_payload("Unauthorized access", "Unauthorized"), status=401)
 
 
+class GraphqlDownloadEvidence(JwtRequiredMixin, HasuraActionView):
+    """
+    Return a download URL or base64-encoded evidence file for authenticated users with proper permissions.
+
+    **Parameters**
+
+    ``input``
+        Dictionary containing the evidence ID and optional returnBase64 flag
+    """
+
+    required_inputs = [
+        "id",
+    ]
+
+    def post(self, request, *args, **kwargs):
+        logger.info(
+            "Received evidence download request from %s",
+            request.META.get("HTTP_X_FORWARDED_FOR", request.META.get("REMOTE_ADDR", None)),
+        )
+
+        evidence_id = self.input["id"]
+        return_base64 = self.input.get("returnBase64", False)
+
+        # Get evidence object
+        try:
+            evidence = Evidence.objects.select_related('finding__report__project', 'report__project').get(pk=evidence_id)
+        except Evidence.DoesNotExist:
+            return JsonResponse(
+                utils.generate_hasura_error_payload("Evidence not found", "EvidenceDoesNotExist"),
+                status=HTTPStatus.NOT_FOUND
+            )
+
+        # Check if user has permission to view the evidence
+        project = evidence.finding.report.project if evidence.finding else evidence.report.project
+        if not project.user_can_view(self.user_obj):
+            return JsonResponse(
+            utils.generate_hasura_error_payload("Forbidden - no access to project", "Unauthorized"),
+            status=HTTPStatus.FORBIDDEN
+            )
+
+        # Return response based on request type
+        if return_base64:
```

Comment:
It's possible this may not be included or may be set `false` when the user has requested `filebase64` in their output. We should ensure the base64 is returned if it is requested (without needing an extra input).